### PR TITLE
Completed tasks

### PR DESCRIPTION
**Pull request by CallumCPP**
Merge dev branch with master. Open for reviewers

### CODE REVIEW

File: CMP1903_A1_2324/Game.cs [hunk 1]
```diff
@@ -16,6 +16,65 @@ internal class Game
...
+        public List<Die> dice = new List<Die>();
+        public int total { get; private set; }
 
+        private int testingIterations, currentIteration;
+        private char choice = ' ';
+
+
+        private bool Condition()
+        {
+            if (testingIterations == 0)
```
Comment (FNASmileys) [inline CMP1903_A1_2324/Game.cs]: braces should be used even when optional

File: CMP1903_A1_2324/Game.cs
Comment (FNASmileys) [inline CMP1903_A1_2324/Game.cs:1]: comments needed

File: CMP1903_A1_2324/Die.cs
Comment (FNASmileys) [inline CMP1903_A1_2324/Die.cs:1]: comments needed

File: CMP1903_A1_2324/Program.cs
Comment (FNASmileys) [inline CMP1903_A1_2324/Program.cs:1]: comments needed

File: CMP1903_A1_2324/Testing.cs
Comment (FNASmileys) [inline CMP1903_A1_2324/Testing.cs:1]: comments needed

File: CMP1903_A1_2324/Die.cs [hunk 2]
```diff
@@ -14,10 +14,14 @@ internal class Die
          */
 
         //Property
-
+        public int value = 0;
```
Comment (FNASmileys) [inline CMP1903_A1_2324/Die.cs]: "Value" not "value" as public variables must be in PascalCase

File: CMP1903_A1_2324/Game.cs [hunk 3]
```diff
@@ -16,6 +16,65 @@ internal class Game
          */
 
         //Methods
+        public List<Die> dice = new List<Die>();
```
Comment (FNASmileys) [inline CMP1903_A1_2324/Game.cs]: public variables must be in PascalCase

File: CMP1903_A1_2324/Game.cs [hunk 4]
```diff
@@ -16,6 +16,65 @@ internal class Game
...
+        public int total { get; private set; }
```
Comment (FNASmileys) [inline CMP1903_A1_2324/Game.cs]: public variables must be in PascalCase >:(

File: CMP1903_A1_2324/Die.cs
Comment (mgoodwin-dev) [inline CMP1903_A1_2324/Die.cs:1]: Code meets functionality requirements
public Variables must be PascalCase, private variables must be _camelCase
Good readability but needs comments to show what the code does

File: CMP1903_A1_2324/Game.cs
Comment (mgoodwin-dev) [inline CMP1903_A1_2324/Game.cs:1]: Code meets functionality requirements
public Variables must be PascalCase, private variables must be _camelCase, braces should be used when optional
Good readability but needs comments to explain the code

File: CMP1903_A1_2324/Program.cs
Comment (mgoodwin-dev) [inline CMP1903_A1_2324/Program.cs:1]: Functionality works as intended although the code will say testing completed whether the user decides to test or not
follows the style guide
Comments are needed for better readability

File: CMP1903_A1_2324/Testing.cs
Comment (mgoodwin-dev) [inline CMP1903_A1_2324/Testing.cs:1]: Functionality works as intended
follows the style guide
Comments are needed for better readability
Some test cases are checked for but could be more thorough testing to check all cases like the amount of dice rolled is correct

**Review by mgoodwin-dev — COMMENTED** [review]
The functionality of the code is met throughout

The style guide is not met throughout, usually when it comes to naming conventions 

Comments are needed throughout the whole project to get better readability aswell as following the style guide fully

Test cases test for the majority of errors and test the die scripts enough times to know it works correctly

**Pull request merged**